projects/sandbox/datagen/datagen/utils/timeslide_waveforms.py

```diff
@@ -0,0 +1,131 @@
+from collections import defaultdict
+from pathlib import Path
+from textwrap import dedent
+from typing import List, Tuple
+
+import h5py
+import numpy as np
+import torch
+
+from ml4gw.spectral import normalize_psd
+
+
+def calc_shifts_required(
+    segments: List[Tuple[int, int]], Tb: float, shift: float
+):
+    """
+    Based off of the lengths of the segments and the
+    amount of data that will need to be sloughed off
+    the ends due to shifting, calculate how many shifts
+    will be required to achieve Tb seconds worth of background
+
+    Args:
+        segments: A list of tuples of the start and stop times of the segments
+        Tb: The amount of background data to generate
+        shift: The increment to shift the data by
+
+    Returns the number of shifts required to achieve Tb seconds of background
+    """
+
+    livetime = sum([stop - start for start, stop in segments])
+    n_segments = len(segments)
+    shifts_required = 0
```

Comment:
Wouldn't save you much, but you could start at `(livetime - 1) // Tb + 1`

Or, you could solve analytically, but that's probably more complicated and less readable.

Reply:
Rather, `(Tb - 1) // livetime +1`

Reply:
Yeah calculating it analytically required solving a quadratic, I got lazy and just did it this way. It's more robust and probably faster analytically so can do that.

Reply:
On second thought, I think this is simpler, shouldn't eat up that much time, and is only done once.

Reply:
Yeah, fully agree, the loop is the way to go

Reply:
One thing I was thinking about today as I was putting the inference stuff together is that as far as I can tell, the main motivation of having the timeslides be the same length was to keep filenames consistent and make it easier to make a correspondence between the files in different shift directories. Since we don't have to do that anymore, do we need to enforce this constraint, or can the 0-shift sequence just be longer than the other sequences?

I guess that will make some headache on dealing with the saved sequences of the _inferred_ data, but if we make the cluster distributions in the inference script as well that largely mitigates this. In fact, there's probably an argument that you don't actually need to save the network outputs at all: given the model weights and the input data you can always reconstruct them (i.e. we could do inference live in the vizapp).

Reply:
I agree, my only concern is we would be breaking many projects (e.g. analysis, vizapp) with this one change , but if they are going to change anyways then might as well rip the band-aid off?

Reply:
I think I'm in favor of ripping it but we can discuss more during our sync today. Would it maybe make sense to make this PR against the `inference-overhaul` branch and have it plug in with those changes, that way we don't have a broken version of `main`?

Reply:
Definitely I realized yesterday I PR'd against the wrong branch